### PR TITLE
[Impeller] distinguish between no clear color and transparent black clear color.

### PR DESCRIPTION
**Pull request by jonahwilliams**
If we clear to transparent black, we're not forcing the pass to be constructed. Change the entity pass API so that we can tell the difference between clearing transparent black and not having a clear color.

In https://github.com/flutter/flutter/issues/139571 , the app is creating a layer that is clearing to a transparent color, which is getting skipped. That invalid texture is fed into a blend which produces either black or magenta error texture.

Fixes https://github.com/flutter/flutter/issues/139571

### CODE REVIEW

File: impeller/entity/entity_pass.cc [hunk 1]
```diff
@@ -342,7 +342,8 @@ bool EntityPass::Render(ContentContext& renderer,
   if (reads_from_onscreen_backdrop) {
     auto offscreen_target =
         CreateRenderTarget(renderer, root_render_target.GetRenderTargetSize(),
-                           GetClearColor(render_target.GetRenderTargetSize()));
+                           GetClearColor(render_target.GetRenderTargetSize())
+                               .value_or(Color::BlackTransparent()));
```
Comment (gaaclarke) [inline impeller/entity/entity_pass.cc:346]: We should pull out this constant or pull out a helper function.

**Review by gaaclarke — APPROVED** [review]
LGTM!

**Comment by gaaclarke** [comment]
Did you intend for the new golden image to draw just white?  We should fix that.

**Comment by jonahwilliams** [comment]
It needs to draw transparent black, for reasons :)

Without this change it is magenta on macOS.

File: impeller/entity/entity_pass.cc (hunk 1 above)
Comment (jonahwilliams) [reply impeller/entity/entity_pass.cc:346]: GetClearColorOrTransparentBlack()?

Comment (gaaclarke) [reply impeller/entity/entity_pass.cc:346]: Here are some options.  I think "OrDefault" is more succinct.  I'm fine with either though.
```c++
GetClearColor(render_target.GetRenderTargetSize())
  .value_or(kDefaultClearColor);
```
```c++
OrDefaultClearColor(
  GetClearColor(render_target.GetRenderTargetSize()))
```
```c++
GetClearColorOrDefault(render_target.GetRenderTargetSize())
```

Comment (jonahwilliams) [reply impeller/entity/entity_pass.cc:346]: Went with OrDefault, Thanks!

**Pull request merged**